libs/MVS/SceneDensify.h

```diff
@@ -0,0 +1,103 @@
+/*
+* DepthMap.h
+*
+* Copyright (c) 2014-2015 SEACAVE
+*
+* Author(s):
+*
+*      cDc <cdc.seacave@gmail.com>
+*
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Affero General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Affero General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+*
+* Additional Terms:
+*
+*      You are required to preserve legal notices and author attributions in
+*      that material or in the Appropriate Legal Notices displayed by works
+*      containing it.
+*/
+
+#ifndef _MVS_SCENEDENSIFY_H_
+#define _MVS_SCENEDENSIFY_H_
+
+// I N C L U D E S /////////////////////////////////////////////////
+
+#include "Common.h"
+#include "Scene.h"
+
+// S T R U C T S ///////////////////////////////////////////////////
+
+namespace MVS {
+// Forward declarations
+class Scene;
+// structure used to compute all depth-maps
+class MVS_API DepthMapsData
+{
+public:
+	DepthMapsData(Scene& _scene);
+	~DepthMapsData();
+
+	bool SelectViews(IIndexArr& images, IIndexArr& imagesMap, IIndexArr& neighborsMap);
+	bool SelectViews(DepthData& depthData);
+	bool InitViews(DepthData& depthData, IIndex idxNeighbor, IIndex numNeighbors);
+	bool InitDepthMap(DepthData& depthData);
+	bool EstimateDepthMap(IIndex idxImage);
+
+	bool RemoveSmallSegments(DepthData& depthData);
+	bool GapInterpolation(DepthData& depthData);
+
+	bool FilterDepthMap(DepthData& depthData, const IIndexArr& idxNeighbors, bool bAdjust=true);
+	void FuseDepthMaps(PointCloud& pointcloud, bool bEstimateColor, bool bEstimateNormal);
+
+protected:
+	static void* STCALL ScoreDepthMapTmp(void*);
+	static void* STCALL EstimateDepthMapTmp(void*);
+	static void* STCALL EndDepthMapTmp(void*);
+
+public:
+  Scene& scene;
+
+  DepthDataArr arrDepthData;
+
+	// used internally to estimate the depth-maps
+	Image8U::Size prevDepthMapSize; // remember the size of the last estimated depth-map
+	Image8U::Size prevDepthMapSizeTrg; // ... same for target image
+  DepthEstimator::MapRefArr coords; // map pixel index to zigzag matrix coordinates
+  DepthEstimator::MapRefArr coordsTrg; // ... same for target image
+};
+
+struct MVS_API DenseDepthMapData {
+  Scene& scene;
+  IIndexArr images;
+  IIndexArr neighborsMap;
```

Comment:
You seem to have a tab alignment issue. It could be fixed by your text/code editor.
You have some tabs with 2 or 4 spaces.

Reply:
My editor uses spaces, the repo seems to have standardized on tabs.

I'll fix it up.

Reply:
I replaced instances of leading spaces with leading tabs in all cases, as seems to be the style of the repository. Resolving this comment.